packages/edit-site/src/components/block-editor/index.js

```diff
@@ -51,28 +58,252 @@ const LAYOUT = {
 	alignments: [],
 };
 
+const FOCUSABLE_ENTITIES = [ 'wp_template_part', 'wp_navigation' ];
+
 export default function BlockEditor() {
-	const { setIsInserterOpened } = useDispatch( editSiteStore );
-	const { storedSettings, templateType, canvasMode, hasPageContentFocus } =
-		useSelect(
-			( select ) => {
-				const {
-					getSettings,
-					getEditedPostType,
-					getCanvasMode,
-					hasPageContentFocus: _hasPageContentFocus,
-				} = unlock( select( editSiteStore ) );
-
-				return {
-					storedSettings: getSettings( setIsInserterOpened ),
-					templateType: getEditedPostType(),
-					canvasMode: getCanvasMode(),
-					hasPageContentFocus: _hasPageContentFocus(),
-				};
-			},
-			[ setIsInserterOpened ]
+	const { templateType, hasPageContentFocus } = useSelect( ( select ) => {
+		const { getEditedPostType, hasPageContentFocus: _hasPageContentFocus } =
+			unlock( select( editSiteStore ) );
+
+		return {
+			templateType: getEditedPostType(),
+			hasPageContentFocus: _hasPageContentFocus(),
+		};
+	}, [] );
+
+	const settings = useSiteEditorSettings( templateType );
+
+	const BlockEditorComponent = getBlockEditorComponent( templateType );
+
+	return (
+		<BlockEditorComponent
+			settings={ settings }
+			templateType={ templateType }
+		>
+			{ hasPageContentFocus && <DisableNonPageContentBlocks /> }
+			<TemplatePartConverter />
+			<SidebarInspectorFill>
+				<BlockInspector />
+			</SidebarInspectorFill>
+
+			<SiteEditorCanvas
+				settings={ settings }
```

Comment:
It's not clear to me why we're passing the "settings" here (It's not introduced in this PR I know). Could probably be avoided by fetching the setting we want in the `SiteEditorCanvas`